source/common/http/http1/codec_impl.cc

```diff
@@ -376,16 +377,80 @@ void ServerConnectionImpl::onEncodeComplete() {
   }
 }
 
+
+  // handlePath computes the correct :authority and :path headers based on 7230#5.7 and 7230#6
+  //
+  bool
+  ServerConnectionImpl::handlePath(HeaderMapImplPtr& headers, bool is_connect) {
+  // TODO(mattwoodyard) - check configuration option if forward proxy is enabled
+  // In practice there will only ever be one scheme delivered for forward proxy
+  // support
+    HeaderString path(Headers::get().Path);
+
+    // The url is relative nothing to do here
+    if (active_request_->request_url_.c_str()[0] == '/' || active_request_->request_url_.c_str()[0] == '*') {
+      headers->addViaMove(std::move(path), std::move(active_request_->request_url_));
+      return true;
+    }
+
+    if (!codec_settings_.allow_connect_ && !codec_settings_.allow_absolute_url_) {
+      return true;
```

Comment:
So I'm curious what the behavior of envoy was for relative URLs prior to this PR if one had a config which just passed everything through to a backend.  Given you have a test I assume fully qualified requests failed before?   I'm always nervous about changing default behavior like (having been bitten by this myself =P) but if you can verify that absolute urls didn't work then that's at least a step in the right direction.